dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/http2/GenericHttp2ServerTransportListener.java

```diff
@@ -187,6 +187,12 @@
         return serverChannelObserver;
     }
 
+    @Override
+    public void onStreamClosed() {
+        // doing on event loop thread
+        getStreamingDecoder().close();
+    }
```

Comment:
**Scalability Issue**: Implementing onStreamClosed to directly close the streaming decoder without considering the state or existence of ongoing operations can lead to abrupt termination of valid processing. This approach might not scale well in scenarios where graceful shutdown or cleanup is necessary for maintaining system stability under load. <br> **Fix**: Ensure that onStreamClosed gracefully handles ongoing operations, possibly by signaling them to complete or terminate in an orderly fashion, before closing the decoder. <br> **Code Suggestion**: 
 ```
 +    @Override
 +    public void onStreamClosed() {
 +        // Signal ongoing operations to complete or terminate in an orderly fashion
 +        if (ongoingOperationsExist()) {
 +            signalOngoingOperationsToCompleteOrTerminate();
 +        }
 +        getStreamingDecoder().close();
 +    }
 ```